x-pack/plugin/sql/sql-action/src/main/java/org/elasticsearch/xpack/sql/action/SqlQueryResponse.java

```diff
@@ -214,17 +214,19 @@ public static XContentBuilder value(XContentBuilder builder, Mode mode, Object v
     }
 
     public static ColumnInfo readColumnInfo(StreamInput in) throws IOException {
-        String table = in.readString();
+        String table = in.readOptionalString();
```

Comment:
In SQL all fields belong to the same schema - we don't really differentiate between different sources and at this point, I think it's fine to use the base name as the table name.

Reply:
> it's fine to use the base name as the table name.

Just to make sure I haven't introduce some naming confusion: the introduced `ColumnInfo`#`baseName` member maps to the field name, not column's/field's index (pattern); i.e. "table".

A bit of a naming background: the JDBC differentiates between a column [label](https://docs.oracle.com/en/java/javase/14/docs/api/java.sql/java/sql/ResultSetMetaData.html#getColumnLabel(int)) and a column [name](https://docs.oracle.com/en/java/javase/14/docs/api/java.sql/java/sql/ResultSetMetaData.html#getColumnName(int)) and only has the notion of a [table name](https://docs.oracle.com/en/java/javase/14/docs/api/java.sql/java/sql/ResultSetMetaData.html#getTableName(int)). [ODBC](https://docs.microsoft.com/en-us/sql/odbc/reference/syntax/sqlcolattribute-function) differentiates between a column label (`SQL_DESC_LABEL`) vs base column name (`SQL_DESC_BASE_COLUMN_NAME`), as well as a table name (`SQL_DESC_TABLE_NAME`) vs a base table name (`SQL_DESC_BASE_TABLE_NAME`).

Since ES/SQL hasn't been so far exposing a column's source (i.e. field name), it's been fine to only have a `ColumnInfo`#`name` for it, but that actually maps on JDBC/ODBC _label_. Since this change is supposed to introduce this differentiation, adding a `label` member and then change all the code to use that, instead of the current `name` seemed an extensive change, so I went with ODBC's nomenclature, i.e. a `baseName`. 
So `name` now maps to JDBC's _label_ and `baseName` to _name_.

But happy to reconsider all this if there's a better way.